ledger/participant-integration-api/src/main/scala/platform/apiserver/services/tracking/QueueBackedTracker.scala

```diff
@@ -47,34 +47,23 @@ private[services] final class QueueBackedTracker(
     )
     logger.trace("Tracking command")
     val trackedPromise = Promise[Either[CompletionFailure, CompletionSuccess]]()
-    queue
-      .offer(Ctx(trackedPromise, submission))
-      .flatMap[Either[TrackedCompletionFailure, CompletionSuccess]] {
-        case QueueOfferResult.Enqueued =>
-          trackedPromise.future.map(
-            _.left.map(completionFailure => QueueCompletionFailure(completionFailure))
-          )
-        case QueueOfferResult.Failure(t) =>
-          toQueueSubmitFailure(
-            errorFactories.SubmissionQueueErrors
-              .failedToEnqueueCommandSubmission("Failed to enqueue")(t)
-          )
-        case QueueOfferResult.Dropped =>
-          toQueueSubmitFailure(errorFactories.bufferFull("The submission ingress buffer is full"))
-        case QueueOfferResult.QueueClosed =>
-          toQueueSubmitFailure(
-            errorFactories.SubmissionQueueErrors.queueClosed("Command service queue")
-          )
-      }
-      .recoverWith {
-        case i: IllegalStateException
-            if i.getMessage == "You have to wait for previous offer to be resolved to send another request" =>
-          toQueueSubmitFailure(errorFactories.bufferFull("The submission ingress buffer is full"))
-        case t =>
-          toQueueSubmitFailure(
-            errorFactories.SubmissionQueueErrors.failedToEnqueueCommandSubmission("Failed")(t)
-          )
-      }
```

Comment:
Don't you want to handle exceptions in `offer` just like previously failed futures?

Reply:
As pointed here: https://github.com/digital-asset/daml/pull/11807#discussion_r754918730, I don't think it's needed. It wouldn't be possible to test it because of that.

Reply:
Ok, I checked the code of the `BoundedSourceQueue` and I see a throw.

Reply:
Ignore, this is not `offer`.

Reply:
I think testing could be done by mocking as we just need to test our reaction to a potential throw. By looking at https://github.com/akka/akka/pull/29770 it looks like you are right and `offer` is not supposed to throw; on the other hand ee don't control `BoundedSourceQueue` and it doesn't seem to mention that it's supposed not to throw, so it may start throwing after an Akka Stream upgrade. If it's not too much work, I think it's worth coding the handling so that at least we start seeing some internal error if this breaks.

Reply:
Fine to me, on my way.